.github/workflows/stackhpc-container-image-build.yml

```
@@ -101,17 +108,23 @@ jobs:
         run: |
           echo "${{ steps.datetime_tag.outputs.datetime_tag }}"
 
+  runner-selection:
+    uses: ./.github/workflows/runner-selector.yml
```

nit: Shouldn't need the leading `./` 
```suggestion
    uses: .github/workflows/runner-selector.yml
```

---

It was necessary to make sure it would run the workflow from the same branch that it is working on, because it didn't exist elsewhere.

---

Do we want to change these or leave them, it works as it is and will always run the ```runner-selector``` workflow from the working branch.

---

I'm not sure I understand why the behaviour would be different. Surely the path is identical for both lines?

---

Just following the GitHub [docs](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow)